module/zfs/zfs_fm.c

```diff
@@ -140,7 +140,10 @@ zfs_is_ratelimiting_event(const char *subclass, vdev_t *vd)
 	return (rc);
 }
 
-static void
+/*
+ * Return 0 if the event was actually posted, return 1 if not.
+ */
+static int
```

Comment:
Making the return type a `boolean_t` would be less misleading if we only care about success and failure.  Otherwise with a return code of `0` I'd expect an `errno` to be returned on failure.